stingray/utils.py

```diff
@@ -149,6 +151,65 @@ def mad(data, c=0.6745, axis=None):
 ]
 
 
+def is_sorted(array):
+    """Check if an array is sorted.
+
+    Checks if an array has extended precision before calling the
+    ``is_sorted`` numba-compiled function.
+
+    Parameters
+    ----------
+    array : iterable
+        The array to be checked
+
+    Returns
+    -------
+    is_sorted : bool
+        True if the array is sorted, False otherwise
+    """
+
+    array = np.asarray(array)
+    # If the array is empty or has only one element, it is sorted
+    if array.size <= 1:
+        return True
+
+    # If Numba is not installed, use numpy's implementation
+    if not HAS_NUMBA:
+        return np.all(np.diff(array) >= 0)
+    # Test if value is compatible with Numba's type system
+    try:
+        _is_sorted_numba(array[:2])
```

Comment:
I'm a bit confused why you're running `_is_sorted_numba` twice, once for the first couple of elements, and then for everything? What's happening here?

Reply:
Oh, is this to check whether the array is extended precision? I feel like there should be a more explicit way to check that rather than just figuring out if Numba fails?

Reply:
It's kind difficult to test whether an array is extended precision, because it can be called in different ways (np.longdouble, np.float128...), different architecture define these with different precision, and I'm not sure I have full control of all possible ways. Also, I wanted this test to be robust even for other kinds of data (not necessarily extended precision floats) that Numba does not accept and I haven't thought about.

Reply:
:+1: